### PR TITLE
CORE-5388 named queries

### PR DESCRIPTION
**Pull request by dickon**

### CODE REVIEW

File: components/virtual-node/entity-processor-service-impl/src/integrationTest/kotlin/net/corda/entityprocessor/impl/tests/PersistenceServiceInternalTests.kt [hunk 1]
```diff
@@ -692,6 +775,36 @@ class PersistenceServiceInternalTests {
         return assertThatResponseIsAList(entityResponse)
     }
 
+    private fun assertNamedQueryDog(params: Map<String, String>, query:String="Dog.summon",
```
Comment (lankydan) [inline components/virtual-node/entity-processor-service-impl/src/integrationTest/kotlin/net/corda/entityprocessor/impl/tests/PersistenceServiceInternalTests.kt]: Can we run this through the formatter, this indentation triggers me 😢

File: components/virtual-node/entity-processor-service-impl/src/main/kotlin/net/corda/entityprocessor/impl/internal/PersistenceServiceInternal.kt [hunk 2]
```diff
@@ -154,4 +156,30 @@ class PersistenceServiceInternal(
...
+    /*
+     * Find all entites that match a named query
+     */
+    fun findWithNamedQuery(
+        serializationService: SerializationService,
+        entityManager: EntityManager,
+        payload: FindWithNamedQuery
+    ): EntityResponse {
+        val query = entityManager.createNamedQuery(payload.queryName)
+        payload.parameters.filter { it.value != null}.forEach { rec ->
```
Comment (lankydan) [inline components/virtual-node/entity-processor-service-impl/src/main/kotlin/net/corda/entityprocessor/impl/internal/PersistenceServiceInternal.kt:187]: Wouldn't `null` values be valid when passed into queries?

**Review by lankydan — COMMENTED** [review]
Dno if I'm meant to be reviewing just yet, but here's one anyway.

File: components/virtual-node/entity-processor-service-impl/src/integrationTest/kotlin/net/corda/entityprocessor/impl/tests/PersistenceServiceInternalTests.kt (hunk 1 above)
Comment (dickon) [reply components/virtual-node/entity-processor-service-impl/src/integrationTest/kotlin/net/corda/entityprocessor/impl/tests/PersistenceServiceInternalTests.kt]: Done

File: components/virtual-node/entity-processor-service-impl/src/main/kotlin/net/corda/entityprocessor/impl/internal/PersistenceServiceInternal.kt (hunk 2 above)
Comment (dickon) [reply components/virtual-node/entity-processor-service-impl/src/main/kotlin/net/corda/entityprocessor/impl/internal/PersistenceServiceInternal.kt:187]: Nulls are not handled by HQL and not allowed in the AVRO format. I added a detailed comment to explain.

**Review by driessamyn — APPROVED** [review]
LGTM

**Pull request merged**